instrumentation/spring/spring-integration-4.1/javaagent/src/test/java/SpringIntegrationAndRabbitTest.java

```diff
@@ -0,0 +1,173 @@
+                                s -> s.isEqualTo("publish")),
+                            satisfies(
+                                MessagingIncubatingAttributes.MESSAGING_MESSAGE_BODY_SIZE,
+                                l -> l.isInstanceOf(Long.class)),
+                            satisfies(
+                                MessagingIncubatingAttributes
+                                    .MESSAGING_RABBITMQ_DESTINATION_ROUTING_KEY,
+                                s -> s.isInstanceOf(String.class))),
+                // spring-cloud-stream-binder-rabbit listener puts all messages into a BlockingQueue
+                // immediately after receiving
+                // that's why the rabbitmq CONSUMER span will never have any child span (and
+                // propagate context, actually)
+                span ->
+                    span.satisfies(
+                            spanData ->
+                                assertThat(spanData.getName())
+                                    .matches("testTopic.anonymous.[-\\w]+ process"))
+                        .hasParent(trace.getSpan(6))
+                        .hasKind(SpanKind.CONSUMER)
+                        .hasAttributesSatisfyingExactly(
+                            satisfies(
+                                NetworkAttributes.NETWORK_PEER_ADDRESS,
+                                s -> s.isIn("127.0.0.1", "0:0:0:0:0:0:0:1", null)),
+                            satisfies(
+                                NetworkAttributes.NETWORK_PEER_PORT,
+                                l -> l.isInstanceOf(Long.class)),
+                            satisfies(
+                                NetworkAttributes.NETWORK_TYPE, s -> s.isIn("ipv4", "ipv6", null)),
+                            equalTo(MessagingIncubatingAttributes.MESSAGING_SYSTEM, "rabbitmq"),
+                            satisfies(
+                                MessagingIncubatingAttributes.MESSAGING_DESTINATION_NAME,
+                                s -> s.isEqualTo("testTopic")),
+                            satisfies(
+                                MessagingIncubatingAttributes.MESSAGING_OPERATION,
+                                s -> s.isEqualTo("process")),
+                            satisfies(
+                                MessagingIncubatingAttributes.MESSAGING_MESSAGE_BODY_SIZE,
+                                l -> l.isInstanceOf(Long.class)),
+                            satisfies(
+                                MessagingIncubatingAttributes
+                                    .MESSAGING_RABBITMQ_DESTINATION_ROUTING_KEY,
+                                s -> s.isInstanceOf(String.class))),
+                // spring-integration will detect that spring-rabbit has already created a consumer
+                // span and back off
+                span ->
+                    span.hasName("testTopic process")
+                        .hasParent(trace.getSpan(6))
```

Comment:
same with this one, its parent is [3](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/2100a1646cbc447cc92533ecb6308a2eb51cc95c/instrumentation/spring/spring-integration-4.1/javaagent/src/test/groovy/SpringIntegrationAndRabbitTest.groovy#L91) in the groovy instead of 6, i guess many of them got swapped